src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Data.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices
+{
+    public sealed partial class PlatformCompatabilityAnalyzer
+    {
+        private enum PlatformAttributeType
+        {
+            None,
+            MinimumOSPlatformAttribute,
+            ObsoletedInOSPlatformAttribute,
+            RemovedInOSPlatformAttribute,
+            TargetPlatformAttribute
+        }
+
+        private readonly struct PlatformAttributeInfo : IEquatable<PlatformAttributeInfo>
+        {
+            public PlatformAttributeType AttributeType { get; }
+            public string PlatformName { get; }
+            public Version Version { get; }
+
+            private PlatformAttributeInfo(PlatformAttributeType attributeType, string platformName, Version version)
+            {
+                AttributeType = attributeType;
+                PlatformName = platformName;
+                Version = version;
+            }
+
+            public static bool TryParsePlatformAttributeInfo(AttributeData osAttribute, out PlatformAttributeInfo parsedAttribute)
+            {
+                if (!osAttribute.ConstructorArguments.IsEmpty &&
+                    osAttribute.ConstructorArguments[0].Type.SpecialType == SpecialType.System_String &&
+                    !osAttribute.ConstructorArguments[0].IsNull &&
+                    !osAttribute.ConstructorArguments[0].Value.Equals(string.Empty) &&
```

Comment:
Can we have `Value` being `null`?

Reply:
Yes if null string passed as constructor argument